### DISCUSSION TITLE
Newbie - can you help me please?

### DISCUSSION THREAD

**Discussion by peter2454** [saleweaver/python-amazon-sp-api#1111] [General]
Hi very new to this library & Python.  Looking for help!

Just trying to connect through to the Amazon API for the first time with the following basic code & query, but am getting an error. The error is:

 _raise MissingCredentials(f'Credentials are missing: {", ".join(required_credentials)}')
sp_api.base.credential_provider.MissingCredentials: Credentials are missing: aws_access_key, aws_secret_key, lwa_app_id, lwa_client_secret_

Can anybody throw any light for me please on what this error actually means? here is my code (pretty much the Quickstart code from the documentation) .   With my actual values for the placeholders (REFRESH_TOKEN etc) of course,:-

--------- Start code --------------------------
```
from datetime import datetime, timedelta
from sp_api.base import Marketplaces
from sp_api.api import Orders
from sp_api.util import throttle_retry, load_all_pages

@throttle_retry()
@load_all_pages()
def load_all_orders(**kwargs):
    """
    A generator function to return all pages, obtained by NextToken
    """
    return Orders().get_orders(**kwargs)

# Set up credentials dictionary
credentials = dict(
    refresh_token='REFRESH_TOKEN',
    lwa_client_secret='LWA_CLIENT_SECRET',
    aws_access_key='AWS_ACCESS_KEY',
    aws_secret_key='AWS_SECRET_KEY',
    role_arn='ROLE_ARN'
)

# Set up the Orders API client with credentials and marketplace
order_client = Orders(credentials=credentials, marketplace=Marketplaces.UK)  


# Main execution
if __name__ == '__main__':
    # Calculate the date 7 days ago
    last_updated_after = (datetime.utcnow() - timedelta(days=7)).isoformat()
    
    # Retrieve and process orders
    for page in load_all_orders(LastUpdatedAfter=last_updated_after):
        for order in page.payload.get('Orders'):
            # Process each order as needed
            print(order)
```
--------------------------End code------------------------

Many thanks!

**Comment by saleweaver** [comment]
You're creating an authenticated order client which you never use. Instead, you're creating a new client (Orders()) in the load all pages function that is not authenticated

**Comment by peter2454** [comment]
Many thanks! That was just the help I needed to understand what was going on. I have adjusted my code and now everything is working as expected! Fantastic help, thank you again.
